### PR TITLE
Draft: Resolve 126 - Mongo: use the new Entity.CollectionName in queries etc.

### LINKED ISSUE #126: Mongo: use the new Entity.CollectionName in queries etc.
In #124 we added an optional `CollectionName` property to `EntityAttribute`. It is however only used as part of the `MongoManager.EnsureIndexes` flows.

We should probably use it in the query layers etc.

### PR DESCRIPTION
**Pull request by IceAndSnow**
Closes #126

### CODE REVIEW

**Comment by IceAndSnow** [comment]
@pingvinen 
This is ready to be reviewed

File: src/mongo/test-functional/MongoDbConnection_GetCollection_Tests.cs [hunk 1]
```diff
@@ -16,19 +18,59 @@ public MongoDbConnection_GetCollection_Tests()
...
+            var actual = connection.GetCollection<DocumentWithCollectionSpecified>("kewl_stuff");
 
-            Assert.IsAssignableFrom<IMongoCollection<Document>>(actual);
+            Assert.IsAssignableFrom<IMongoCollection<DocumentWithCollectionSpecified>>(actual);
             Assert.Equal("kewl_stuff", actual.CollectionNamespace.CollectionName);
         }
 
         [Fact]
-        public void GetCollection_usesNameOfDocumentType_ifNotGivenACollectionName()
+        public void GetCollection_usesCollectionNameDefinedInEntityAttribute_ifParamIsNull()
```
Comment (pingvinen) [inline src/mongo/test-functional/MongoDbConnection_GetCollection_Tests.cs:28]: **praise**: nice to see that you also updated the name of the test :)

That is something that many tend to forget :)

**Review by pingvinen — APPROVED** [review]
Nice work

**Comment by pingvinen** [comment]
I just removed the "breaking change" label, as I do not see anything that breaks the public parts of the library.

**Pull request merged**